src/ripple/ledger/impl/View.cpp

```diff
@@ -810,19 +805,28 @@ describeOwnerDir(AccountID const& account)
     };
 }
 
-std::pair<TER, bool>
+boost::optional<std::uint64_t>
 dirAdd (ApplyView& view,
-    std::uint64_t&                          uNodeDir,
     Keylet const&                           dir,
     uint256 const&                          uLedgerIndex,
+    bool                                    strictOrder,
     std::function<void (SLE::ref)>          fDescriber,
     beast::Journal j)
```

Comment:
There are two `dirAdd` functions:
- The existing legacy, free function, which this PR deprecates. I _had_ to insert the `bool` parameter to it, temporarily, but the function is going away.
- The new low-level `dirAdd` function (which is `private` to the class) and accepts a lambda for the insertion strategy. Users only invoke `ApplyView::dirInsert` or `ApplyView::dirAppend`.